### DISCUSSION TITLE
(Ubuntu) Where is the executable after build?

### DISCUSSION THREAD

**Discussion by JerkyChew899** [Chia-Network/bladebit#31] [Q&A]
Running Ubuntu 20.04. I cloned the repo to the root of my system and built it in /bladebit. But I don't know where the executable is - There is no executable bladebit file on my system. Where should I be looking to run the command?

**Comment by harold-b** [comment]
It will be in (relative to project directory) .bin/release/bladebit

or for ARM:
.bin/release.arm/bladebit

**Reply by JerkyChew899** [reply]
I must have done something wrong. I followed the instructions (I'm on Ubuntu 21.04 on x86) and ran all the commands from within a /bladebit directory I created. After the git clone and installation, I only have the following folders:

> drwxr-xr-x  9 root root  4096 Jul 23 23:23 ./
drwxr-xr-x 22 root root  4096 Jul 23 23:21 ../
-rw-r--r--  1 root root   368 Jul 23 23:21 bladebit.natvis
-rw-r--r--  1 root root  1543 Jul 23 23:21 bladebit.sln
-rw-r--r--  1 root root 24405 Jul 23 23:21 bladebit.vcxproj
-rw-r--r--  1 root root 20725 Jul 23 23:21 bladebit.vcxproj.filters
-rwxr-xr-x  1 root root  1080 Jul 23 23:21 build-bls*
drwxr-xr-x  5 root root  4096 Jul 23 23:21 builds/
drwxr-xr-x  9 root root  4096 Jul 23 23:21 .git/
-rw-r--r--  1 root root   125 Jul 23 23:21 .gitignore
-rw-r--r--  1 root root   141 Jul 23 23:21 .gitmodules
drwxr-xr-x  4 root root  4096 Jul 23 23:22 lib/
-rw-r--r--  1 root root 34454 Jul 23 23:21 LICENSE
-rw-r--r--  1 root root  3573 Jul 23 23:21 Makefile
drwxr-xr-x  3 root root  4096 Jul 23 23:21 modules/
-rw-r--r--  1 root root 42184 Jul 23 23:21 NOTICE
drwxr-xr-x 11 root root  4096 Jul 23 23:23 .obj/
-rw-r--r--  1 root root  2826 Jul 23 23:21 project.js
-rw-r--r--  1 root root  4451 Jul 23 23:21 README.md
drwxr-xr-x 13 root root  4096 Jul 23 23:21 src/
-rwxr-xr-x  1 root root   525 Jul 23 23:21 sync*
drwxr-xr-x  2 root root  4096 Jul 23 23:21 .vscode/

**Reply by JerkyChew899** [reply]
I've found the source of the issue. I went and started over and noticed this error on compilation:
`src/platform/linux/SysHost_Linux.cpp:9:10: fatal error: numa.h: No such file or directory
    9 | #include <numa.h>
      |          ^~~~~~~~
compilation terminated.
`
It's looking for numa.h which I didn't have - The numactl-dev package your instructions refer to isn't the name of the numa dev libraries in Ubuntu. Did some googling and resolved the dependency by running sudo apt-get install -y libnuma-dev first. Then I re-re-recompiled and I now have a .bin directory.

**Reply by cemalefendi** [reply]
hi

i have same problem.  i cant create bin folder.

a@a-System-Product-Name:~/bladebit$ make clean && make -j$(nproc --all)
Compiling for linux with g++
src/b3/blake3_avx2_x86-64_unix.S
src/b3/blake3_avx512_x86-64_unix.S
src/b3/blake3_dispatch.c
src/b3/blake3_portable.c
src/b3/blake3_sse41_x86-64_unix.S
src/b3/blake3.c
src/bech32/segwit_addr.c
src/fse/debug.c
src/fse/entropy_common.c
src/fse/fse_compress.c
src/fse/fse_decompress.c
src/fse/hist.c
src/io/FileStream.cpp
src/main.cpp
src/memplot/DbgHelper.cpp
src/memplot/MemPhase1.cpp
src/memplot/MemPhase2.cpp
src/memplot/MemPhase3.cpp
src/memplot/MemPhase4.cpp
src/memplot/MemPlotter.cpp
src/pch.cpp
src/platform/linux/SysHost_Linux.cpp
src/platform/unix/FileStream_Unix.cpp
src/PlotContext.cpp
src/PlotWriter.cpp
src/pos/chacha8.cpp
src/SysHost.cpp
src/threading/Semaphore.cpp
src/threading/Thread.cpp
src/threading/ThreadPool.cpp
src/Util.cpp
src/util/Log.cpp
a@a-System-Product-Name:~/bladebit$ 

i cant give any error. how can i solve this problem.

**Reply by harold-b** [reply]
If your build completed successfully, you should find the specified directories above:

> It will be in (relative to project directory) .bin/release/bladebit
> 
> or for ARM:
> .bin/release.arm/bladebit

**Reply by regonsite** [reply]
> If your build completed successfully, you should find the specified directories above:
> 
> > It will be in (relative to project directory) .bin/release/bladebit
> > or for ARM:
> > .bin/release.arm/bladebit

Have same problem, during installation there is no errors shown, but .bin directory doesn't exist at all.
